test/e2e/live_iso_test.go

```diff
@@ -0,0 +1,69 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("When testing live iso [live-iso]", func() {
+	liveIsoTest()
+})
+
+func liveIsoTest() {
+	var (
+		liveIsoImageURL = e2eConfig.GetVariable("LIVE_ISO_IMAGE")
+	)
+	BeforeEach(func() {
+		validateGlobals(specName)
+		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
+		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+	})
+	It("Should update the bmh with live iso", func() {
+		bootstrapClient := bootstrapClusterProxy.GetClient()
+
+		bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
+		Expect(err).NotTo(HaveOccurred(), "Error getting bmhs")
+
+		var isoBmh bmov1alpha1.BareMetalHost
+		Expect(bootstrapClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: bmhs[0].Name}, &isoBmh)).To(Succeed())
```

Comment:
I think we can simplify this a bit.
```suggestion
		isoBmh := bmhs[0]
```
Maybe add an Èxpect` to check that it is not `nil`. But the point is that we don't have to fetch the BMH twice.